extensions/smallrye-health/deployment/src/main/java/io/quarkus/smallrye/health/deployment/SmallRyeHealthProcessor.java

```diff
@@ -67,16 +87,35 @@ void build(SmallRyeHealthTemplate template, RecorderContext recorder,
             BuildProducer<AdditionalBeanBuildItem> additionalBean,
             BuildProducer<BeanDefiningAnnotationBuildItem> beanDefiningAnnotation) throws IOException {
 
+        //String livenessPath = health.healthPath + health.livenessRelPath;
+        //String readinessPath = health.healthPath + health.readinessRelPath;
+
         feature.produce(new FeatureBuildItem(FeatureBuildItem.SMALLRYE_HEALTH));
 
-        // Register the servlet
+        // Register the health servlet
         ServletBuildItem servletBuildItem = ServletBuildItem.builder("health", SmallRyeHealthServlet.class.getName())
-                .addMapping(health.path).build();
+                .addMapping(health.healthPath).build();
         servlet.produce(servletBuildItem);
 
+        // Register the liveness servlet
+        ServletBuildItem liveServletBuildItem = ServletBuildItem.builder("liveness", SmallRyeHealthServlet.class.getName())
+                .addMapping(health.healthPath + health.livenessRelPath).build();
+        servlet.produce(liveServletBuildItem);
+
+        // Register the readiness servlet
+        ServletBuildItem readyServletBuildItem = ServletBuildItem.builder("readiness", SmallRyeHealthServlet.class.getName())
```

Comment:
You point to the exact same servlet for liveness and readiness. It feels weird?

Reply:
probably an error

Reply:
Not only it feels weird. It's imho wrong - this servlet always reports an overall health - all available checks. See https://github.com/smallrye/smallrye-health/blob/master/tck/src/main/java/io/smallrye/health/SmallRyeHealthServlet.java#L19.

Reply:
It should be probably `SmallRyeLivenessServlet` and `SmallRyeReadinessServlet`.